src/MongoDB.Driver.Core/Core/Operations/UpdateOpcodeOperation.cs

```diff
@@ -216,6 +216,10 @@ private WriteConcernResult ExecuteProtocol(IChannelHandle channel, CancellationT
             {
                 throw new NotSupportedException("OP_UPDATE does not support arrayFilters.");
             }
+            if (_request.Hint != null)
+            {
+                throw new NotSupportedException("Unacknowledged OP_UPDATE does not support hints.");
```

Comment:
The proper error message should be:

    throw new NotSupportedException("OP_UPDATE does not support hints.");

It is not only Unacknowledged OP_UPDATE that does not support hints, it is **ALL** OP_UPDATEs whether acknowledged or not.

Reply:
Done.